homeassistant/components/smartthings/light.py

```diff
@@ -0,0 +1,204 @@
+
+        # State is set optimistically in the commands above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state(True)
+
+    async def async_turn_off(self, **kwargs) -> None:
+        """Turn the light off."""
+        # Switch/transition
+        if self._supported_features & SUPPORT_TRANSITION \
+                and ATTR_TRANSITION in kwargs:
+            await self.set_level(0, int(kwargs[ATTR_TRANSITION]))
+        else:
+            await self._device.switch_off(set_status=True)
+
+        # State is set optimistically in the commands above, therefore update
+        # the entity state ahead of receiving the confirming push updates
+        self.async_schedule_update_ha_state(True)
+
+    async def async_update(self):
+        """Call when the device is refreshed."""
+        from pysmartthings.device import Capability
+
+        self._supported_features = 0
+        # Brightness and transition
+        if Capability.switch_level in self._device.capabilities:
+            self._supported_features |= \
+                SUPPORT_BRIGHTNESS | SUPPORT_TRANSITION
+            self._brightness = convert_scale(
+                self._device.status.level, 100, 255)
+        # Color Temperature
+        if Capability.color_temperature in self._device.capabilities:
+            self._supported_features |= SUPPORT_COLOR_TEMP
+            self._color_temp = color_util.color_temperature_kelvin_to_mired(
+                self._device.status.color_temperature)
+        # Color
+        if Capability.color_control in self._device.capabilities:
+            self._supported_features |= SUPPORT_COLOR
+            self._hs_color = (
+                convert_scale(self._device.status.hue, 100, 360),
+                self._device.status.saturation
+            )
+
+    async def set_color(self, hs_color):
+        """Set the color of the device."""
+        hue = convert_scale(float(hs_color[0]), 360, 100)
+        hue = max(min(hue, 100.0), 0.0)
+        saturation = max(min(float(hs_color[1]), 100.0), 0.0)
+        await self._device.set_color(
+            hue, saturation, set_status=True)
+
+    async def set_color_temp(self, value: float):
+        """Set the color temperature of the device."""
+        kelvin = color_util.color_temperature_mired_to_kelvin(value)
+        kelvin = max(min(kelvin, 30000.0), 1.0)
+        await self._device.set_color_temperature(
+            kelvin, set_status=True)
+
+    async def set_level(self, brightness: int, transition: int):
+        """Set the brightness of the light over transition."""
+        level = int(convert_scale(brightness, 255, 100, 0))
+        # Due to rounding, set level to 1 (one) so we don't inadvertently
+        # turn off the light when a low brightness is set.
+        level = 1 if level == 0 and brightness > 0 else level
+        level = max(min(level, 100), 0)
+        duration = max(int(transition), 0)
```

Comment:
Transition is already validated as a float and min value is 0.

Reply:
Removed!